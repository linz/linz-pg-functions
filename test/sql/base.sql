\set ECHO none
--------------------------------------------------------------------------------
--
-- Copyright 2016 Crown copyright (c)
-- Land Information New Zealand and the New Zealand Government.
-- All rights reserved
--
-- This software is released under the terms of the new BSD license. See the 
-- LICENSE file for more information.
--
--------------------------------------------------------------------------------
-- Provide unit testing for LINZ misc PostgreSQL functions using pgTAP
--------------------------------------------------------------------------------
\set QUIET true
\set VERBOSITY terse
\pset format unaligned
\pset tuples_only true

SET client_min_messages TO WARNING;

--BEGIN;

\i sql/create_table_polygon_grid.sql

CREATE EXTENSION pgtap;

SELECT plan(1);

SELECT has_function( 'public', 'create_table_polygon_grid', ARRAY['name','name','name', 'double precision', 'double precision']  );

SELECT * FROM finish();

--ROLLBACK;

