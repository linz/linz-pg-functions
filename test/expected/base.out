\set ECHO none
1..9
ok 1 - Function public.create_table_polygon_grid(name, name, name, double precision, double precision) should exist
ok 2 - threw Table public.test does not exists
ok 3 - threw Table public.test geometry column fake does not exists
ok 4 - threw Table 'public'.test column g does not contain any polygons
ok 5 - threw Table 'public'.test does not have a unique non-composite, non-null primary key
ok 6 - create_table_polygon_grid returns test_grid
ok 7 - Sum of geometry areas is 400
ok 8 - No grid geometry has area bigger than 100
ok 9 - Symdifference between source geom and union of grid geoms is empty
